packages/terra-table/src/Table.jsx

```diff
@@ -0,0 +1,326 @@
+
+      offsetArray.push(cumulativeOffset);
+      const lastPinnedColumnIndex = pinnedColumns.length - 1;
+
+      renderedColumns.slice(0, lastPinnedColumnIndex).forEach((pinnedColumn) => {
+        cumulativeOffset += pinnedColumn.width;
+        offsetArray.push(cumulativeOffset);
+      });
+    }
+
+    setPinnedColumnOffsets(offsetArray);
+
+    const currentSortedColumn = renderedColumns.find((column) => !!column.sortIndicator);
+    if (sortedColumn.current) {
+      const previousSortedColumn = { ...sortedColumn.current };
+
+      if (previousSortedColumn.id !== currentSortedColumn.id
+      || previousSortedColumn.sortIndicator !== currentSortedColumn.sortIndicator) {
+        setCellAriaLiveMessage(intl.formatMessage(
+          { id: 'Terra.table.sort-direction-changed' },
+          {
+            column: currentSortedColumn.displayName,
+            sortDirection: currentSortedColumn.sortIndicator,
+          },
+        ));
+      }
+    }
+
+    sortedColumn.current = currentSortedColumn;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [renderedColumns]);
+
+  // -------------------------------------
+
+  // -------------------------------------
+  // event handlers
+
+  const handleColumnSelect = useCallback((columnId) => {
+    if (onColumnSelect) {
+      onColumnSelect(columnId);
+    }
+  }, [onColumnSelect]);
+
+  const onResizeMouseDown = useCallback((event, index, resizeColumnWidth) => {
+    // Store current table and column values for resize calculations
+    tableWidth.current = tableRef.current.offsetWidth;
+    activeColumnPageX.current = event.pageX;
+    activeColumnWidth.current = resizeColumnWidth;
+
+    // Set the active index to the selected column
+    setActiveIndex(index);
+  }, []);
+
+  const onMouseMove = (event) => {
+    if (activeIndex == null) {
+      return;
+    }
+
+    // Ensure the new column width falls within the range of the minimum and maximum values
+    const diffX = event.pageX - activeColumnPageX.current;
+    const { minimumWidth, maximumWidth } = renderedColumns[activeIndex];
+    const newColumnWidth = Math.min(Math.max(activeColumnWidth.current + diffX, minimumWidth), maximumWidth);
+
+    // Update the width for the column in the state variable
+    const newColumns = [...renderedColumns];
+    newColumns[activeIndex].width = newColumnWidth;
+    setRenderedColumns(newColumns);
+
+    // Update the column and table width
+    tableRef.current.style.width = `${tableWidth + (newColumnWidth - activeColumnWidth.current)}px`;
+  };
+
+  const onMouseUp = () => {
+    if (onColumnResize) {
+      onColumnResize(renderedColumns[activeIndex].id, renderedColumns[activeIndex].width);
+    }
+    // Remove active index
+    setActiveIndex(null);
+  };
+
+  // -------------------------------------
+
+  return (
+    <div
+      className={cx('table-container')}
+      ref={tableContainerRef}
+      tabIndex={gridContext.role === 'grid' ? 0 : -1}
```

Comment:
The tabIndex for the table container would not be 0 for a grid control.  Also, why would we set the value to -1 for a table role.  Who is writing code to set focus to this container?

Reply:
Removed as discussed per sync.